Typos CI check crashed

https://github.com/JuliaLang/julia/actions/runs/7276276842/job/19825773753?pr=45641

from PR

https://github.com/JuliaLang/julia/pull/45641

at commit

7f9b382c07d030b24b25f35583f8934f3890e56b

<details>
<summary>full log</summary>

```
2023-12-20T13:48:36.6026830Z Requested labels: ubuntu-latest
2023-12-20T13:48:36.6027120Z Job defined at: JuliaLang/julia/.github/workflows/Typos.yml@refs/pull/45641/merge
2023-12-20T13:48:36.6027243Z Waiting for a runner to pick up this job...
2023-12-20T13:48:37.2697125Z Job is waiting for a hosted runner to come online.
2023-12-20T13:48:39.9452391Z Job is about to start running on the hosted runner: GitHub Actions 13 (hosted)
2023-12-20T13:48:42.5186929Z Current runner version: '2.311.0'
2023-12-20T13:48:42.5209666Z ##[group]Operating System
2023-12-20T13:48:42.5210296Z Ubuntu
2023-12-20T13:48:42.5210756Z 22.04.3
2023-12-20T13:48:42.5211102Z LTS
2023-12-20T13:48:42.5211435Z ##[endgroup]
2023-12-20T13:48:42.5211847Z ##[group]Runner Image
2023-12-20T13:48:42.5212289Z Image: ubuntu-22.04
2023-12-20T13:48:42.5212688Z Version: 20231217.2.0
2023-12-20T13:48:42.5213708Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20231217.2/images/ubuntu/Ubuntu2204-Readme.md
2023-12-20T13:48:42.5215164Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20231217.2
2023-12-20T13:48:42.5216024Z ##[endgroup]
2023-12-20T13:48:42.5216443Z ##[group]Runner Image Provisioner
2023-12-20T13:48:42.5216940Z 2.0.321.1
2023-12-20T13:48:42.5217275Z ##[endgroup]
2023-12-20T13:48:42.5218118Z ##[group]GITHUB_TOKEN Permissions
2023-12-20T13:48:42.5219613Z Metadata: read
2023-12-20T13:48:42.5220110Z ##[endgroup]
2023-12-20T13:48:42.5223419Z Secret source: None
2023-12-20T13:48:42.5224194Z Prepare workflow directory
2023-12-20T13:48:42.5840330Z Prepare all required actions
2023-12-20T13:48:42.5995935Z Getting action download info
2023-12-20T13:48:42.8923637Z Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
2023-12-20T13:48:43.1299473Z Complete job name: Check for new typos
2023-12-20T13:48:43.2232795Z ##[group]Run actions/checkout@v4
2023-12-20T13:48:43.2233334Z with:
2023-12-20T13:48:43.2233852Z   persist-credentials: false
2023-12-20T13:48:43.2234355Z   repository: JuliaLang/julia
2023-12-20T13:48:43.2235003Z   token: ***
2023-12-20T13:48:43.2235453Z   ssh-strict: true
2023-12-20T13:48:43.2235836Z   clean: true
2023-12-20T13:48:43.2236238Z   sparse-checkout-cone-mode: true
2023-12-20T13:48:43.2236793Z   fetch-depth: 1
2023-12-20T13:48:43.2237198Z   fetch-tags: false
2023-12-20T13:48:43.2237610Z   show-progress: true
2023-12-20T13:48:43.2238034Z   lfs: false
2023-12-20T13:48:43.2238385Z   submodules: false
2023-12-20T13:48:43.2238767Z   set-safe-directory: true
2023-12-20T13:48:43.2239285Z ##[endgroup]
2023-12-20T13:48:48.0031446Z Syncing repository: JuliaLang/julia
2023-12-20T13:48:48.0033659Z ##[group]Getting Git version info
2023-12-20T13:48:48.0034508Z Working directory is '/home/runner/work/julia/julia'
2023-12-20T13:48:48.0035766Z [command]/usr/bin/git version
2023-12-20T13:48:48.0036437Z git version 2.43.0
2023-12-20T13:48:48.0038257Z ##[endgroup]
2023-12-20T13:48:48.0163043Z Temporarily overriding HOME='/home/runner/work/_temp/a2210fa6-bdb4-4a37-8b03-c4e779d75d59' before making global git config changes
2023-12-20T13:48:48.0165941Z Adding repository directory to the temporary git global config as a safe directory
2023-12-20T13:48:48.0168445Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/julia/julia
2023-12-20T13:48:48.0200538Z Deleting the contents of '/home/runner/work/julia/julia'
2023-12-20T13:48:48.0207671Z ##[group]Initializing the repository
2023-12-20T13:48:48.0209902Z [command]/usr/bin/git init /home/runner/work/julia/julia
2023-12-20T13:48:48.0268384Z hint: Using 'master' as the name for the initial branch. This default branch name
2023-12-20T13:48:48.0270360Z hint: is subject to change. To configure the initial branch name to use in all
2023-12-20T13:48:48.0272102Z hint: of your new repositories, which will suppress this warning, call:
2023-12-20T13:48:48.0273428Z hint: 
2023-12-20T13:48:48.0274541Z hint: 	git config --global init.defaultBranch <name>
2023-12-20T13:48:48.0275525Z hint: 
2023-12-20T13:48:48.0276663Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2023-12-20T13:48:48.0278565Z hint: 'development'. The just-created branch can be renamed via this command:
2023-12-20T13:48:48.0279805Z hint: 
2023-12-20T13:48:48.0280535Z hint: 	git branch -m <name>
2023-12-20T13:48:48.0281642Z Initialized empty Git repository in /home/runner/work/julia/julia/.git/
2023-12-20T13:48:48.0288877Z [command]/usr/bin/git remote add origin https://github.com/JuliaLang/julia
2023-12-20T13:48:48.0324086Z ##[endgroup]
2023-12-20T13:48:48.0325694Z ##[group]Disabling automatic garbage collection
2023-12-20T13:48:48.0327197Z [command]/usr/bin/git config --local gc.auto 0
2023-12-20T13:48:48.0356672Z ##[endgroup]
2023-12-20T13:48:48.0358106Z ##[group]Setting up auth
2023-12-20T13:48:48.0367315Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-12-20T13:48:48.0398267Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-12-20T13:48:48.0685230Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-12-20T13:48:48.0713562Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-12-20T13:48:48.0944453Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2023-12-20T13:48:48.0978101Z ##[endgroup]
2023-12-20T13:48:48.0979310Z ##[group]Fetching the repository
2023-12-20T13:48:48.0991236Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +80cb6eb21fa15897587e6359ebd05a93570cead4:refs/remotes/pull/45641/merge
2023-12-20T13:48:50.2003236Z From https://github.com/JuliaLang/julia
2023-12-20T13:48:50.2005190Z  * [new ref]         80cb6eb21fa15897587e6359ebd05a93570cead4 -> pull/45641/merge
2023-12-20T13:48:50.2028297Z ##[endgroup]
2023-12-20T13:48:50.2028909Z ##[group]Determining the checkout info
2023-12-20T13:48:50.2031054Z ##[endgroup]
2023-12-20T13:48:50.2031623Z ##[group]Checking out the ref
2023-12-20T13:48:50.2035813Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/45641/merge
2023-12-20T13:48:50.3710264Z Note: switching to 'refs/remotes/pull/45641/merge'.
2023-12-20T13:48:50.3711097Z 
2023-12-20T13:48:50.3711731Z You are in 'detached HEAD' state. You can look around, make experimental
2023-12-20T13:48:50.3713008Z changes and commit them, and you can discard any commits you make in this
2023-12-20T13:48:50.3714263Z state without impacting any branches by switching back to a branch.
2023-12-20T13:48:50.3715028Z 
2023-12-20T13:48:50.3715529Z If you want to create a new branch to retain commits you create, you may
2023-12-20T13:48:50.3716787Z do so (now or later) by using -c with the switch command. Example:
2023-12-20T13:48:50.3717426Z 
2023-12-20T13:48:50.3717766Z   git switch -c <new-branch-name>
2023-12-20T13:48:50.3718280Z 
2023-12-20T13:48:50.3718461Z Or undo this operation with:
2023-12-20T13:48:50.3718740Z 
2023-12-20T13:48:50.3718883Z   git switch -
2023-12-20T13:48:50.3719090Z 
2023-12-20T13:48:50.3719394Z Turn off this advice by setting config variable advice.detachedHead to false
2023-12-20T13:48:50.3719898Z 
2023-12-20T13:48:50.3720381Z HEAD is now at 80cb6eb Merge 7f9b382c07d030b24b25f35583f8934f3890e56b into f8cd1eb4bee56058a5f12383ac49990fc1c98ee7
2023-12-20T13:48:50.3724189Z ##[endgroup]
2023-12-20T13:48:50.3761414Z [command]/usr/bin/git log -1 --format='%H'
2023-12-20T13:48:50.3785599Z '80cb6eb21fa15897587e6359ebd05a93570cead4'
2023-12-20T13:48:50.3794170Z ##[group]Removing auth
2023-12-20T13:48:50.3797313Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-12-20T13:48:50.3824643Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-12-20T13:48:50.4056358Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-12-20T13:48:50.4076371Z http.https://github.com/.extraheader
2023-12-20T13:48:50.4084742Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-12-20T13:48:50.4114870Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-12-20T13:48:50.4344445Z ##[endgroup]
2023-12-20T13:48:50.4682777Z ##[group]Run git fetch --depth=1 origin master
2023-12-20T13:48:50.4683586Z [36;1mgit fetch --depth=1 origin master[0m
2023-12-20T13:48:50.4684209Z [36;1mOLD_FILES=$(git diff-index --name-only --diff-filter=ad FETCH_HEAD)[0m
2023-12-20T13:48:50.4684926Z [36;1mNEW_FILES=$(git diff-index --name-only --diff-filter=d FETCH_HEAD)[0m
2023-12-20T13:48:50.4685562Z [36;1m[0m
2023-12-20T13:48:50.4685959Z [36;1mmkdir -p "/home/runner/work/_temp/typos"[0m
2023-12-20T13:48:50.4686409Z [36;1mRELEASE_ASSET_URL="$([0m
2023-12-20T13:48:50.4686977Z [36;1m  gh api /repos/crate-ci/typos/releases/latest \[0m
2023-12-20T13:48:50.4687843Z [36;1m  --jq '."assets"[] | select(."name" | test("^typos-.+-x86_64-unknown-linux-musl\\.tar\\.gz$")) | ."browser_download_url"'[0m
2023-12-20T13:48:50.4688610Z [36;1m)"[0m
2023-12-20T13:48:50.4689269Z [36;1mwget --secure-protocol=TLSv1_3 --max-redirect=1 --retry-on-host-error --retry-connrefused --tries=3 \[0m
2023-12-20T13:48:50.4690296Z [36;1m  --quiet --output-document=- "${RELEASE_ASSET_URL}" \[0m
2023-12-20T13:48:50.4690960Z [36;1m  | tar -xz -C "/home/runner/work/_temp/typos" ./typos[0m
2023-12-20T13:48:50.4691516Z [36;1m"/home/runner/work/_temp/typos/typos" --version[0m
2023-12-20T13:48:50.4691980Z [36;1m[0m
2023-12-20T13:48:50.4692759Z [36;1mecho -n $NEW_FILES | xargs "/home/runner/work/_temp/typos/typos" --format json >> /home/runner/work/_temp/new_typos.jsonl || true[0m
2023-12-20T13:48:50.4693590Z [36;1mgit checkout FETCH_HEAD -- $OLD_FILES[0m
2023-12-20T13:48:50.4694431Z [36;1mecho -n $OLD_FILES | xargs "/home/runner/work/_temp/typos/typos" --format json >> /home/runner/work/_temp/old_typos.jsonl  || true[0m
2023-12-20T13:48:50.4695272Z [36;1m[0m
2023-12-20T13:48:50.4695593Z [36;1mpython -c '[0m
2023-12-20T13:48:50.4695915Z [36;1mimport sys, json[0m
2023-12-20T13:48:50.4696354Z [36;1mold = set()[0m
2023-12-20T13:48:50.4696748Z [36;1mwith open(sys.argv[1]) as old_file:[0m
2023-12-20T13:48:50.4697161Z [36;1m  for line in old_file:[0m
2023-12-20T13:48:50.4697677Z [36;1m    old.add(json.loads(line)["typo"])[0m
2023-12-20T13:48:50.4698122Z [36;1mclean = True[0m
2023-12-20T13:48:50.4698474Z [36;1mwith open(sys.argv[2]) as new_file:[0m
2023-12-20T13:48:50.4698998Z [36;1m  for line in new_file:[0m
2023-12-20T13:48:50.4699402Z [36;1m    new = json.loads(line)[0m
2023-12-20T13:48:50.4699788Z [36;1m    if new["typo"] not in old:[0m
2023-12-20T13:48:50.4700517Z [36;1m      if len(new["typo"]) > 6: # Short typos might be false positives. Long are probably real.[0m
2023-12-20T13:48:50.4701399Z [36;1m        clean = False[0m
2023-12-20T13:48:50.4701961Z [36;1m      print("::warning file={},line={},col={}::perhaps \"{}\" should be \"{}\".".format([0m
2023-12-20T13:48:50.4702760Z [36;1m        new["path"], new["line_num"], new["byte_offset"],[0m
2023-12-20T13:48:50.4703348Z [36;1m        new["typo"], " or ".join(new["corrections"])))[0m
2023-12-20T13:48:50.4704203Z [36;1msys.exit(1 if not clean else 0)' "/home/runner/work/_temp/old_typos.jsonl" "/home/runner/work/_temp/new_typos.jsonl"[0m
2023-12-20T13:48:50.4770143Z shell: /usr/bin/bash -e {0}
2023-12-20T13:48:50.4770681Z env:
2023-12-20T13:48:50.4771282Z   GH_TOKEN: ***
2023-12-20T13:48:50.4771610Z ##[endgroup]
2023-12-20T13:48:50.9402452Z From https://github.com/JuliaLang/julia
2023-12-20T13:48:50.9403338Z  * branch            master     -> FETCH_HEAD
2023-12-20T13:48:50.9407235Z  * [new branch]      master     -> origin/master
2023-12-20T13:48:55.9163349Z typos-cli 1.16.25
2023-12-20T13:48:56.0278319Z Previous HEAD position was 80cb6eb Merge 7f9b382c07d030b24b25f35583f8934f3890e56b into f8cd1eb4bee56058a5f12383ac49990fc1c98ee7
2023-12-20T13:48:56.0280927Z HEAD is now at f8cd1eb `invperm` for immutable arrays (#52546)
2023-12-20T13:48:56.8490251Z Traceback (most recent call last):
2023-12-20T13:48:56.8491138Z   File "<string>", line 6, in <module>
2023-12-20T13:48:56.8491789Z KeyError: 'typo'
2023-12-20T13:48:56.8595261Z ##[error]Process completed with exit code 1.
2023-12-20T13:48:56.8697006Z Post job cleanup.
2023-12-20T13:48:56.9425795Z [command]/usr/bin/git version
2023-12-20T13:48:56.9465898Z git version 2.43.0
2023-12-20T13:48:56.9509668Z Temporarily overriding HOME='/home/runner/work/_temp/449a673f-6940-4cc9-bf43-311663c2c1c9' before making global git config changes
2023-12-20T13:48:56.9511590Z Adding repository directory to the temporary git global config as a safe directory
2023-12-20T13:48:56.9515716Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/julia/julia
2023-12-20T13:48:56.9550396Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-12-20T13:48:56.9582724Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-12-20T13:48:56.9824798Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-12-20T13:48:56.9855901Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-12-20T13:48:57.0318211Z Cleaning up orphan processes
```

</details>

--- a/.github/workflows/Typos.yml
+++ b/.github/workflows/Typos.yml
@@ -26,7 +26,7 @@ jobs:
           # This is necessary because the typos command interprets the
           # empty string as "check all files" rather than "check no files".
           if [ -z "$NEW_FILES" ]; then
-            echo "All eddited files were deleted. Skipping typos check."
+            echo "All edited files were deleted. Skipping typos check."
             exit 0
           fi
 

--- a/.github/workflows/Typos.yml
+++ b/.github/workflows/Typos.yml
@@ -54,12 +54,14 @@ jobs:
           old = set()
           with open(sys.argv[1]) as old_file:
             for line in old_file:
-              old.add(json.loads(line)["typo"])
+              j = json.loads(line)
+              if j["type"] == "typo":
+                old.add(j["typo"])
           clean = True
           with open(sys.argv[2]) as new_file:
             for line in new_file:
               new = json.loads(line)
-              if new["typo"] not in old:
+              if new["type"] == "typo" and new["typo"] not in old:
                 if len(new["typo"]) > 6: # Short typos might be false positives. Long are probably real.
                   clean = False
                 print("::warning file={},line={},col={}::perhaps \"{}\" should be \"{}\".".format(

--- a/.github/workflows/Typos.yml
+++ b/.github/workflows/Typos.yml
@@ -23,6 +23,13 @@ jobs:
           OLD_FILES=$(git diff-index --name-only --diff-filter=ad FETCH_HEAD)
           NEW_FILES=$(git diff-index --name-only --diff-filter=d FETCH_HEAD)
 
+          # This is necessary because the typos command interprets the
+          # empty string as "check all files" rather than "check no files".
+          if [ -z "$NEW_FILES" ]; then
+            echo "All eddited files were deleted. Skipping typos check."
+            exit 0
+          fi
+
           mkdir -p "${{ runner.temp }}/typos"
           RELEASE_ASSET_URL="$(
             gh api /repos/crate-ci/typos/releases/latest \
@@ -35,7 +42,12 @@ jobs:
 
           echo -n $NEW_FILES | xargs "${{ runner.temp }}/typos/typos" --format json >> ${{ runner.temp }}/new_typos.jsonl || true
           git checkout FETCH_HEAD -- $OLD_FILES
-          echo -n $OLD_FILES | xargs "${{ runner.temp }}/typos/typos" --format json >> ${{ runner.temp }}/old_typos.jsonl  || true
+          if [ -z "$OLD_FILES" ]; then
+            touch "${{ runner.temp }}/old_typos.jsonl" # No old files, so no old typos.
+          else
+            echo -n $OLD_FILES | xargs "${{ runner.temp }}/typos/typos" --format json >> ${{ runner.temp }}/old_typos.jsonl || true
+          fi
+
 
           python -c '
           import sys, json

